tests/posix/common/src/semaphore.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <errno.h>
+#include <pthread.h>
+#include <semaphore.h>
+
+#define STACK_SIZE 1024
+
+sem_t sema;
+void *dummy_sem;
+
+static K_THREAD_STACK_DEFINE(stack, STACK_SIZE);
+
+static void *child_func(void *p1)
+{
+	zassert_equal(sem_post(dummy_sem), -1, "sem_post of"
```

Comment:
Good call on the dummy semaphore check, however you can move this to the start of the `test_posix_semaphore` function; else we test this twice when calling the child thread again. We only need the successful `sem_post`, don't we?

Reply:
Why can't we call the same thread entry function while creating 2 threads? Anyways it not adding any overhead in terms of execution time right?

Reply:
I was not talking about calling the thread entry function twice. I was talking about the code from lines 21 to 23 being executed twice.